dlt/extract/decorators.py

```diff
@@ -297,6 +297,7 @@ def resource(
     columns: TTableHintTemplate[TAnySchemaColumns] = None,
     primary_key: TTableHintTemplate[TColumnNames] = None,
     merge_key: TTableHintTemplate[TColumnNames] = None,
+    merge_config: TMergeConfig = None,
```

Comment:
@jorritsandbrink @sh-rp maybe instead of doing this, we could extend `write_disposition` to
``
write_disposition: TTableHintTemplate[Union[TWriteDisposition, TMergeConfig, TReplaceConfig]] = None
```
exactly what we do with `schema_contract` below? then we can support short hand strings and full definitions. and still use the same parameter
in that case I'd rename TMergeConfig to TMergeDispositionConfig

Reply:
and we also handle the replace strategies in there? (not in the pr but later)? i think it's a good idea

Reply:
Extended `write_disposition` in https://github.com/dlt-hub/dlt/pull/1168/commits/8f4d4cee6e59e37962963670bf2d23a1d1454e99 and https://github.com/dlt-hub/dlt/pull/1168/commits/c8d84d88a8fc8e221cb3b60c8930be77be67009f